packages/dd-trace/src/opentelemetry/metrics/periodic_metric_reader.js

```diff
@@ -0,0 +1,532 @@
+   * @param {number} value - The value to aggregate
+   * @param {Attributes} attributes - The attributes of the metric
+   * @param {string} attrKey - The attribute key
+   * @param {number} timestamp - The timestamp of the measurement
+   * @param {string} stateKey - The state key
+   * @param {Map<string, any>} cumulativeState - The cumulative state of the metrics
+   */
+  #aggregateSum (metric, value, attributes, attrKey, timestamp, stateKey, cumulativeState) {
+    if (!cumulativeState.has(stateKey)) {
+      cumulativeState.set(stateKey, {
+        value: 0,
+        startTime: metric.temporality === TEMPORALITY.CUMULATIVE ? this.#startTime : timestamp
+      })
+    }
+
+    const state = cumulativeState.get(stateKey)
+    state.value += value
+
+    const dataPoint = this.#findOrCreateDataPoint(metric, attributes, attrKey, () => ({
+      startTimeUnixNano: state.startTime,
+      timeUnixNano: timestamp,
+      value: 0
+    }))
+
+    dataPoint.value = state.value
+    dataPoint.timeUnixNano = timestamp
+  }
+
+  /**
+   * Overwrites the last recorded value for a given metric or
+   * creates a new data point if it doesn't exist.
+   *
+   * @param {AggregatedMetric} metric - The metric to aggregate a last value for
+   * @param {number} value - The value to aggregate
+   * @param {Attributes} attributes - The attributes of the metric
+   * @param {string} attrKey - The attribute key
+   * @param {number} timestamp - The timestamp of the measurement
+   */
+  #aggregateLastValue (metric, value, attributes, attrKey, timestamp) {
+    const dataPoint = this.#findOrCreateDataPoint(metric, attributes, attrKey, () => ({
+      timeUnixNano: timestamp,
+      value: 0
+    }))
+
+    dataPoint.value = value
+    dataPoint.timeUnixNano = timestamp
+  }
+
+  /**
+   * Aggregates histogram values by distributing them into buckets.
+   * Tracks count, sum, min, max, and per-bucket counts and creates
+   * a new data point if it doesn't exist.
+   *
+   * @param {AggregatedMetric} metric - The metric to aggregate a histogram for
+   * @param {number} value - The value to aggregate
+   * @param {Attributes} attributes - The attributes of the metric
+   * @param {string} attrKey - The attribute key
+   * @param {number} timestamp - The timestamp of the measurement
+   * @param {string} stateKey - The state key
+   * @param {Map<string, any>} cumulativeState - The cumulative state of the metrics
+   * @returns {void}
+   */
+  #aggregateHistogram (metric, value, attributes, attrKey, timestamp, stateKey, cumulativeState) {
+    if (!cumulativeState.has(stateKey)) {
+      cumulativeState.set(stateKey, {
+        count: 0,
+        sum: 0,
+        min: Infinity,
+        max: -Infinity,
+        bucketCounts: new Array(DEFAULT_HISTOGRAM_BUCKETS.length + 1).fill(0),
+        startTime: metric.temporality === TEMPORALITY.CUMULATIVE ? this.#startTime : timestamp
+      })
+    }
+
+    const state = cumulativeState.get(stateKey)
```

Comment:
Nit

Ideally, it would just get the state if it was defined. If it was not defined, we could combine the handling to just set the first value.

That would be a tad faster.